documentation/resource_spotlight.md

```diff
@@ -0,0 +1,87 @@
+spotlight
+===
+
+Use the **spotlight** resource to manage the metadata indexing state for disk volumes. This
+will primarily affect the ability to search volume contents with the macOS Spotlight feature.
+Under the hood, a **spotlight** resource executes the `mdutil` command in the `metadata_util`
+library.
+
+[Learn more about Spotlight](https://support.apple.com/en-us/HT204014).
+
+[Learn more about the `mdutil` command](https://developer.apple.com/legacy/library/documentation/Darwin/Reference/ManPages/man1/mdutil.1.html).
+
+Syntax
+------
+
+The most basic usage of a **spotlight** resource block declares a disk volume as the name property
+to **enable** metadata indexing:
+
+```ruby
+spotlight '/'
+```
+
+The full syntax for all of the properties available to the **spotlight** resource is:
+
+```ruby
+spotlight 'volume name' do
+  volume                      String # defaults to 'volume name' if not specified
+  indexed                     TrueClass, FalseClass # defaults to TrueClass if not specified
```

Comment:
`indexing` feels a little vague and doesn't immediately strike me as a boolean property. I think making it explicit might make it clearer, especially if it has a default property. How about something like `indexing_disabled` and have it default to `false`?

Reply:
IMO this is fine. In my head, I usually read these properties as is it`indexed?`.

Reply:
@americanhanko `indexing` isn't the property name - it's `indexed`.

Reply:
Oops! Thanks for that correction.

What I'm getting at is that by naming the property `indexed`, it's as if there is a question mark there: `indexed?` -- that would make sense given the ruby convention for methods that return a boolean value. However, since we can't use a question mark in a property name, I thought it would be better to make it explicit: `indexing_enabled` with a default value of `false` (or vice versa).

Conversely, it turns out boolean properties across Chef (core, community or otherwise) resources do not share a consistent naming strategy, and in fact follow a similar pattern of "leaving out the question mark". Either way works for me 👍